### DISCUSSION TITLE
Still attempt upgrades on non-deployed releases

### DISCUSSION THREAD

**Discussion by JadeHayes** [helmfile/helmfile#1243] [Ideas]
Currently, the "helmfile apply" command attempts to run a "helm diff" between the existing release in the cluster and the proposed release. The decision to run "helm upgrade" is based on whether the diff shows a change. While this is a sensible check for "deployed" releases, it poses a challenge for failed releases.

In the case of failed releases (and potentially other states), if the diff hasn't changed (even though the release never succeeded), "helmfile apply" concludes successfully. This results in the worst possible outcome as it appears to finish successfully, completely ignoring the failed release.

Here is a PR for a proposed solution: https://github.com/helmfile/helmfile/pull/1043

**Comment by yxxhero** [comment]
@mumoshu WDYT?

**Comment by steved** [comment]
@yxxhero @mumoshu just bumping this up, curious if you have thoughts!
